code-challenges.test.js

```diff
@@ -26,27 +36,129 @@ const numbersArray2 = [24, 27, 30, 33, 36]
+describe("capitalArray", () => {
+    it("returns a generic greeting", () => {
+        const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
+        // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
+        const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
+        // Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
+      expect(capitalArray(randomNouns1)).toEqual(["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"])
+      expect(capitalArray(randomNouns2)).toEqual(["Temperature", "Database", "Chopsticks", "Mango"])
+    })
+  })
 
 // b) Create the function that makes the test pass.
 
-// Pseudo code:
\ No newline at end of file
+const capitalArray = (capWords) => {
+    return capWords.map(value => {
+        return value.charAt(0).toUpperCase() + value.slice(1)
+    })
+}
+
+// Pseudo code:
+// input: a single array
+// output: an array with all words capitalized
+// since the array is going to be the the same length with changed values i need to use map
+// we did something similar in the lecture with every other letter needing to be capitalized, so i think i'm going to have to use the same approach.
+// did some googling, because i knew there has to be a built in method to capitalize the first letter of the word and i did see something that should probably work but i wanted to try something else first.
+// const capitalArray = (capWords) => {
+//     return capWords.map(value => {
+//         return value.toUpperCase([0])
+//     })
+// }
+// -----> failed
+// Array [
+//     -   "Streetlamp",
+//     -   "Potato",
+//     -   "Teeth",
+//     -   "Conclusion",
+//     -   "Nephew",
+//     +   "STREETLAMP",
+//     +   "POTATO",
+//     +   "TEETH",
+//     +   "CONCLUSION",
+//     +   "NEPHEW",
+//       ]
+// i thought that maybe i would be able to reference that i wanted the [0] index in each value to be capitalized, but i think i'm gonna just use the way i saw it done on my google search.
+// const capitalArray = (capWords) => {
+//     return capWords.map(value => {
+//         return value.charAt(0).toUpperCase()
+//     })
+// }
+// ----> failed
+// Array [
+//     -   "Streetlamp",
+//     -   "Potato",
+//     -   "Teeth",
+//     -   "Conclusion",
+//     -   "Nephew",
+//     +   "S",
+//     +   "P",
+//     +   "T",
+//     +   "C",
+//     +   "N",
+//       ]
+// technically on the right track, but i need the whole word.
+// google said to use + word.slice(1), but i couldn't rememeber exactly what slice does so i'm just gonna look that up now.
+// passed but i'm almost a little confused as to why it worked. i'm assuming its because using .slice at index 1 is going to input the rest of the word in the value in the array. but on the syllabus if you put .slice(1) it would bring up the rest of the array starting at potato
+// would really appreciate an explaination so i can better understand why this worked.
```

Comment:
Basically, the approach here is handling the first letter in the word and then handling the rest of the word. First `value.charAt(0)` will return the first letter and then uppercase it. 
But we need the rest of the word - the second letter (or first index) through the end of the word.
`value.slice(1)` is creating a subset of the word. What gets passed in is the index of the start of the subset. If you don't pass another argument it will default to the length (or the end of the word).